cumulus_etl/loaders/fhir/export_log.py

```diff
@@ -97,3 +100,186 @@ def _find(self, root: store.Root) -> str:
                 return filenames[log_files[0]]
             case _:
                 raise self.MultipleLogs("Multiple log.*.ndjson files found")
+
+
+class BulkExportLogWriter:
+    """Writes a standard log bulk export file."""
+
+    def __init__(self, root: store.Root):
+        self.root = root
+        self._export_id = str(uuid.uuid4())
+        self._filename = root.joinpath("log.ndjson")
+        self._num_files = 0
+        self._num_resources = 0
+        self._num_bytes = 0
+        self._start_time = None
+
+    def _event(self, event_id: str, detail: dict, *, timestamp: datetime.datetime = None) -> None:
+        timestamp = timestamp or common.datetime_now(local=True)
+        if self._start_time is None:
+            self._start_time = timestamp
+
+        # We open the file anew for each event because:
+        # a) logging should be flushed often to disk
+        # b) it makes the API of the class easier by avoiding a context manager
+        with self.root.fs.open(self._filename, "a", encoding="utf8") as f:
```

Comment:
We currently require `--export-to` folders to be local, FYI. (Largely because the MS deid tool needs a local dir, and we were operating directly on the export dir.)

With this PR, where we now copy the export data to a local folder for the MS tool (because the MS tool would choke on the log file), we _could_ explore relaxing that requirement.

And thus, the log file could be on an S3 bucket. This code here is already using the `fsspec` API, so it could write to S3 (if it transparently supports the "append" write mode...?)

I will say, for my own purposes, I like to export locally and inspect the results for sanity & a lack of errors before "promoting" the results to S3. But maybe that's me treating S3 as special, when we could just write results to it, and delete them if they are bad.